Building a Pre-built SDK Monolithic configuration of a project with all gems enabled fails

**Describe the bug**
Pre-built SDK Monolithic release build of a project with all gems configured fails at the end of the build, resulting in the <o3de-install>\bin\Windows\release\Monolithic not being created. It should be noted that performing the same release on a Standard Template project works without issues. 

Near the end of the build there are multiple LNK errors printed:
```
Creating library C:/build/o3de/build/windows_mono2/lib/release/NewProjectAG.GameLauncher.lib and object C:/build/o3de/build/windows_mono2/lib/release/NewProjectAG.GameLauncher.exp
unity_0_cxx.obj : error LNK2019: unresolved external symbol CreateModuleClass_Gem_CrashReporting referenced in function CreateStaticModules [C:\build\o3de\build\windows_mono2\Code\LauncherUnified\Ne
wProjectAG.GameLauncher.vcxproj]
unity_0_cxx.obj : error LNK2019: unresolved external symbol CreateModuleClass_Gem_MultiplayerCompression referenced in function CreateStaticModules [C:\build\o3de\build\windows_mono2\Code\LauncherUn
ified\NewProjectAG.GameLauncher.vcxproj]
unity_0_cxx.obj : error LNK2019: unresolved external symbol CreateModuleClass_Gem_ScriptCanvasDeveloper referenced in function CreateStaticModules [C:\build\o3de\build\windows_mono2\Code\LauncherUni
fied\NewProjectAG.GameLauncher.vcxproj]
C:\build\o3de\build\windows_mono2\bin\release\NewProjectAG.GameLauncher.exe : fatal error LNK1120: 3 unresolved externals [C:\build\o3de\build\windows_mono2\Code\LauncherUnified\NewProjectAG.GameLau
ncher.vcxproj]
...
Creating library C:/build/o3de/build/windows_mono2/lib/release/NewProjectAG.ServerLauncher.lib and object C:/build/o3de/build/windows_mono2/lib/release/NewProjectAG.ServerLauncher.exp
unity_0_cxx.obj : error LNK2019: unresolved external symbol CreateModuleClass_Gem_CrashReporting referenced in function CreateStaticModules [C:\build\o3de\build\windows_mono2\Code\LauncherUnified\Ne
wProjectAG.ServerLauncher.vcxproj]
unity_0_cxx.obj : error LNK2019: unresolved external symbol CreateModuleClass_Gem_MultiplayerCompression referenced in function CreateStaticModules [C:\build\o3de\build\windows_mono2\Code\LauncherUn
ified\NewProjectAG.ServerLauncher.vcxproj]
unity_0_cxx.obj : error LNK2019: unresolved external symbol CreateModuleClass_Gem_ScriptCanvasDeveloper referenced in function CreateStaticModules [C:\build\o3de\build\windows_mono2\Code\LauncherUni
fied\NewProjectAG.ServerLauncher.vcxproj]
C:\build\o3de\build\windows_mono2\bin\release\NewProjectAG.ServerLauncher.exe : fatal error LNK1120: 3 unresolved externals [C:\build\o3de\build\windows_mono2\Code\LauncherUnified\NewProjectAG.Serve
rLauncher.vcxproj]
```

Please refer to the Command Line log fragment attached in the **Command Line log** section for more information.

**Steps to reproduce**
Steps to reproduce the behavior:
1. Configure and build a Pre-built SDK engine.
2. Create a new project **with all of the available Gems enabled.**
3. Build the project.
4. Launch the project and ensure that the Asset Processor processes all of the assets.
5. Create a level.
6. Close the Engine, Asset Processor, and the Project Manager.
7. Perform the Pre-built SDK Monolithic release build (as described on the https://www.o3de.org/docs/user-guide/packaging/windows-release-builds/ documentation page) on the project created in previous steps.

**Expected behavior**
The release build is successful.

**Actual behavior**
The release build does not finish and errors are printed to the Command Line.

**Command Line log**
[CMDLog.txt](https://github.com/o3de/o3de/files/8124609/CMDLog.txt)

**Found in Branch**
Development (a3ea07f)

**Desktop**
 - Device: PC
 - OS: Windows
 - Version 10
 - CPU AMD Ryzen 5 3600
 - GPU NVIDIA GeForce RTX 2060 SUPER
 - Memory 16GB

--- a/Gems/AtomTressFX/CMakeLists.txt
+++ b/Gems/AtomTressFX/CMakeLists.txt
@@ -31,8 +31,6 @@ if(PAL_TRAIT_BUILD_HOST_TOOLS)
                 Code
                 External
                 External/Code/src
-            PUBLIC
-                Code/Include
         BUILD_DEPENDENCIES
             PRIVATE
                 AZ::AzCore
@@ -56,8 +54,6 @@ if(PAL_TRAIT_BUILD_HOST_TOOLS)
                 Code
                 External
                 External/Code/src
-            PUBLIC
-                Code/Include
         BUILD_DEPENDENCIES
             PRIVATE
                 AZ::AzCore
@@ -79,8 +75,6 @@ else()
                 Code
                 External
                 External/Code/src
-            PUBLIC
-                Code/Include
         BUILD_DEPENDENCIES
             PRIVATE
                 AZ::AzCore
@@ -103,8 +97,6 @@ else()
                 Code
                 External
                 External/Code/src
-            PUBLIC
-                Code/Include
         BUILD_DEPENDENCIES
             PRIVATE
                 AZ::AzCore
@@ -130,8 +122,6 @@ if(PAL_TRAIT_BUILD_HOST_TOOLS)
                 Code
                 External
                 External/Code/src
-            PUBLIC
-                Code/Include
         BUILD_DEPENDENCIES
             PUBLIC
                 AZ::AssetBuilderSDK
@@ -148,8 +138,6 @@ if(PAL_TRAIT_BUILD_HOST_TOOLS)
                 Code
                 External
                 External/Code/src
-            PUBLIC
-                Code/Include
         BUILD_DEPENDENCIES
             PRIVATE
                 Gem::AtomTressFX.Builders.Static

--- a/Gems/CrashReporting/Code/Source/GameCrashHandler.cpp
+++ b/Gems/CrashReporting/Code/Source/GameCrashHandler.cpp
@@ -14,6 +14,7 @@
 #include <AzCore/Component/ComponentApplicationBus.h>
 #include <AzCore/IO/FileIO.h>
 #include <AzCore/IO/SystemFile.h>
+#include <AzCore/Module/Module.h>
 
 namespace CrashHandler
 {
@@ -90,3 +91,5 @@ namespace CrashHandler
         return returnPath;
     }
 }
+
+AZ_DECLARE_MODULE_CLASS(Gem_CrashReporting, AZ::Module)

--- a/Gems/MultiplayerCompression/Code/Source/MultiplayerCompressionModule.cpp
+++ b/Gems/MultiplayerCompression/Code/Source/MultiplayerCompressionModule.cpp
@@ -41,7 +41,4 @@ namespace MultiplayerCompression
     };
 }
 
-// DO NOT MODIFY THIS LINE UNLESS YOU RENAME THE GEM
-// The first parameter should be GemName_GemIdLower
-// The second should be the fully qualified name of the class above
-AZ_DECLARE_MODULE_CLASS(MultiplayerCompression_1d353c8ca3c74ed193fd6c6783ae41cc, MultiplayerCompression::MultiplayerCompressionModule)
+AZ_DECLARE_MODULE_CLASS(Gem_MultiplayerCompression, MultiplayerCompression::MultiplayerCompressionModule)

--- a/Gems/ScriptCanvasDeveloper/Code/Game/Source/ScriptCanvasDeveloperGem.cpp
+++ b/Gems/ScriptCanvasDeveloper/Code/Game/Source/ScriptCanvasDeveloperGem.cpp
@@ -42,4 +42,4 @@ namespace ScriptCanvasDeveloper
     }
 }
 
-AZ_DECLARE_MODULE_CLASS(Gem_ScriptCanvasDeveloperGem, ScriptCanvasDeveloper::ScriptCanvasDeveloperModule)
+AZ_DECLARE_MODULE_CLASS(Gem_ScriptCanvasDeveloper, ScriptCanvasDeveloper::ScriptCanvasDeveloperModule)

